### PR TITLE
DEVHUB-195: Bump Deps and Fix Lingering Peer Dependency Warnings

### PR DESCRIPTION
**Pull request by jestapinski**
[Staging Link](https://docs-mongodbcom-staging.corp.mongodb.com/master/devhub/jordanstapinski/js/fix-peer-deps/)
[JIRA Ticket](https://jira.mongodb.org/browse/DEVHUB-195)

This PR updates a majority of DevHub dependencies, and removes some old peer dependency warnings called out in the linked ticket.

There are still some peer dependency warnings that show when running `npm install`. These are related to outdated dependencies in Gatsby (responsible for `@typescript-eslint/eslint-plugin@2.34.0`, `@typescript-eslint/parser@2.34.0`, `eslint-config-react-app@5.2.1`, and `gatsby-interface@0.0.166`). At this point, any dependency warnings due to dependencies we directly install have been resolved.

Breaking changes found included storybook and react-player. Storybook required a change in the config to load the theme earlier on, whereas react-player required a new callback to be specified called `onReady`. This field triggers when the player is ready to go, so we should wait for it before beginning to use the player. This only was relevant for the podcast player, since we toggle the `playing` prop.

### To Verify

- Check that articles still look okay
- Check that videos and podcasts still play off of the learn page
- Basically nothing should look out of place

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -25,73 +25,78 @@
...
-    "babel-preset-gatsby": "^0.4.7",
-    "cypress": "^4.7.0",
+    "babel-preset-gatsby": "^0.5.9",
+    "core-js": "^3.6.5",
+    "cypress": "^5.1.0",
     "enzyme": "^3.11.0",
-    "enzyme-adapter-react-16": "^1.15.2",
+    "enzyme-adapter-react-16": "^1.15.4",
     "enzyme-to-json": "^3.5.0",
-    "eslint": "^7.1.0",
```
Comment (jestapinski) [inline package.json:84]: For example, three of the remaining seven warnings involve `eslint`:

```
npm WARN @typescript-eslint/eslint-plugin@2.34.0 requires a peer of eslint@^5.0.0 || ^6.0.0 but none is installed. You must install peer dependencies yourself.
npm WARN @typescript-eslint/parser@2.34.0 requires a peer of eslint@^5.0.0 || ^6.0.0 but none is installed. You must install peer dependencies yourself.
npm WARN eslint-config-react-app@5.2.1 requires a peer of eslint@6.x but none is installed. You must install peer dependencies yourself.
```

Instead of downgrading `eslint` on our end, I think it makes more sense to wait for gatsby to upgrade its dependencies and then just upgrade gatsby again when it is ready.

The other warnings are similar for gatsby, react, and react-dom. Just checking out these packages, they still function well and newer versions have (mostly) bumped their peer dependency requirements

**Review by tahiya-mongo — APPROVED** [review]
Looked through the staging link. Nothing looks out of place to me.

**Review by jbouzi12 — APPROVED** [review]
Checked this out and it looks like articles, videos and podcasts are all working as expected.

**Comment by jestapinski** [comment]
Right on, thanks all!
